### PR TITLE
Managed spends

### PR DESCRIPTION
**Pull request by fbrcode**

### CODE REVIEW

File: contracts/tokens/Grid.sol [hunk 1]
```diff
@@ -5,60 +5,69 @@ import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20CappedUp
...
 		_mint(msg.sender, INITIAL_SUPPLY);
-    }
+	}
+
+	/// @notice define token contract administrators
+	/// @dev setup administrator in a way that only one of the 3 admins can perform token actions
+	/// @param _admin1 1st administrator
+	/// @param _admin2 2nd administrator
+	/// @param _admin3 3rd administrator
+	function setGridAdminstrators(
```
Comment (provsalt) [inline contracts/tokens/Grid.sol:65]: Unused function

File: contracts/treasury/Treasury.sol [hunk 2]
```diff
@@ -2,20 +2,51 @@
...
-contract Treasury is Ownable {
 	event TokenDepositEvent(address indexed depositorAddress, address indexed tokenContractAddress, uint256 amount);
 
-	function depositToken(address _token, uint256 _amount) public {
+	/// @notice define tresury contract administrators
+	/// @dev setup administrator in a way that only one of the 3 admins can perform treasury actions
+	/// @param _admin1 1st administrator
+	/// @param _admin2 2nd administrator
+	/// @param _admin3 3rd administrator
+	function setTreasuryAdminstrators(
```
Comment (provsalt) [inline contracts/treasury/Treasury.sol:28]: Perhaps use an array of admins?

File: contracts/treasury/Treasury.sol [hunk 3]
```diff
@@ -24,15 +55,15 @@ contract Treasury is Ownable {
...
 	}
 
-	function withdrawTokens(address tokenAddress, uint256 amount) public onlyOwner {
+	function withdrawTokens(address tokenAddress, uint256 amount) public definedAdmins onlyAdmin {
 		require(address(tokenAddress) != address(0), "Invalid address");
 		IERC20 tokenContract = IERC20(tokenAddress);
 		uint256 tokenBalance = tokenContract.balanceOf(address(this));
 		require(tokenBalance >= amount, "Insufficient token balance");
-		tokenContract.transfer(owner(), amount);
+		tokenContract.transfer(msg.sender, amount);
```
Comment (provsalt) [inline contracts/treasury/Treasury.sol]: require the return value to be true

File: package.json [hunk 4]
```diff
@@ -75,7 +76,7 @@
 		"mocha": "^9.2.2",
 		"node-watch": "^0.7.3",
 		"prettier": "^2.6.2",
-		"prettier-plugin-solidity": "^1.0.0-dev.21",
+		"prettier-plugin-solidity": "^1.0.0-beta.19",
```
Comment (provsalt) [inline package.json]: Is the downgrade necessary?

**Review by provsalt — CHANGES_REQUESTED** [review]

File: contracts/tokens/Grid.sol (hunk 1 above)
Comment (fbrcode) [reply contracts/tokens/Grid.sol:65]: ok, adjusting that

File: package.json (hunk 4 above)
Comment (fbrcode) [reply package.json]: it is not... reverting

File: contracts/treasury/Treasury.sol (hunk 2 above)
Comment (fbrcode) [reply contracts/treasury/Treasury.sol:28]: will change it in the next iteration

File: contracts/account/Administrator.sol [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+	// future: implement voting triad (or other governance option)
+	address[3] public admin;
+
+	/// @notice define token contract administrators
+	/// @dev setup administrator in a way that only one of the 3 admins can perform token actions
+	/// @param _admin array of administators
+	function setAdminstrators(
+		address[3] memory _admin
+	) public {
+		require(_admin[0] != _admin[1] && _admin[0] != _admin[2] && _admin[1] != _admin[2], "Admins must be different addresses");
```
Comment (provsalt) [inline contracts/account/Administrator.sol]: This could be changed to a modifier like onlyAdmin

**Review by provsalt — CHANGES_REQUESTED** [review]
Please write hardhat tests and deploy scripts for the treasury contract before I merge this in. Otherwise, looks good.

File: contracts/account/Administrator.sol (hunk 5 above)
Comment (fbrcode) [reply contracts/account/Administrator.sol]: I did some adjustment preventing the 2nd call

**Review by provsalt — APPROVED** [review]

**Pull request merged**